### DISCUSSION TITLE
QTime/Qinput only output 24 hour time

### DISCUSSION THREAD

**Discussion by DustinShack** [quasarframework/quasar#16668] [General - Components / Directives / etc] [answered]
As an example of my issue, on your documents page here : https://quasar.dev/vue-components/time#with-qinput

You have an example of a QInput working with a QTime, however, the QTime component GUI is in standard time, but the Qinput is in 24hour time. This seems to be because QTime always OUTPUTS 24 hour time. Is there a built in "Quasar" solution to this? What would you suggest I do so that QTime outputs standard time(with AM/PM) to the QInput?

**Comment by metalsadman** [comment, marked as answer]
https://codepen.io/metalsadman/pen/mdvoavp?editors=101
ref:
https://quasar.dev/vue-components/time#model-mask
https://quasar.dev/vue-components/input#mask
https://quasar.dev/vue-components/input#internal-validation

as for the input validation rule, implement your own. `update: added rule`
